### PR TITLE
Bst

### PR DESCRIPTION
**Pull request by corinnelhh**

### CODE REVIEW

File: BST/bst.py [hunk 1]
```diff
@@ -47,6 +46,53 @@ def insert(self, val):
...
+                tmp = tmp._left
+            elif val > tmp._data:
+                tmp_parent = tmp
+                tmp = tmp._right
+            else:
+                return tmp, tmp_parent
+            if tmp is None:
+                raise IndexError
+
+    def delete_node(self, val):
```
Comment (tsnaomi) [inline BST/bst.py:64]: This method is just a tad too complex, and also should be called delete(),

Comment (DryTuna) [inline BST/bst.py:64]: would you like to take a look at our updated version using dictionary as
the placeholder =) ???

https://github.com/corinnelhh/data-structures/blob/debug/BST/duy_bst.py

Comment (DryTuna) [inline BST/bst.py:64]: -.- dammet sent to wrong person, sorry Corinne

Comment (tsnaomi) [inline BST/bst.py:64]: Hey Duy! I'll happily take a look at it =)

Would you mind signing onto the IRC so I can ask you a few questions?

**Pull request merged**